### PR TITLE
feat(condo): DOMA-5456 added analytics for dropdown filters

### PR DESCRIPTION
**Pull request by Alllex202**
plus refactored and typing has been added

### CODE REVIEW

File: apps/condo/domains/common/components/Table/Filters.tsx [hunk 1]
```diff
@@ -141,75 +193,71 @@ export const getGQLSelectFilterDropdown = (
...
+            innerPickerProps.value = [dayjs(selectedKeys[0]), dayjs(selectedKeys[1])]
         }
 
         return (
             <FilterContainer clearFilters={handleClear}
                 showClearButton={selectedKeys && selectedKeys.length > 0}
                 style={containerStyles}
             >
-                <DateRangePicker {...pickerProps} {...props}/>
+                <DateRangePicker  {...outerPickerProps} {...innerPickerProps} />
```
Comment (nomerdvadcatpyat) [inline apps/condo/domains/common/components/Table/Filters.tsx:260]:
```suggestion
                <DateRangePicker {...outerPickerProps} {...innerPickerProps} />
```

**Review by nomerdvadcatpyat — APPROVED** [review]

File: apps/condo/domains/common/components/Table/Filters.tsx [hunk 2]
```diff
@@ -45,37 +68,67 @@ export const getTextFilterDropdown = (placeholder: string, containerStyles?: CSS
                 style={containerStyles}
             >
                 <Input
-                    placeholder={placeholder}
-                    value={selectedKeys}
+                    {...inputProps}
+                    value={selectedKeys as any}
```
Comment (dkoviazin) [inline apps/condo/domains/common/components/Table/Filters.tsx:72]: Why any? Input already has a type for value property

File: apps/condo/domains/common/components/Table/Filters.tsx [hunk 3]
```diff
@@ -141,75 +193,71 @@ export const getGQLSelectFilterDropdown = (
                     style={containerStyles}
                 >
                     <GraphQlSearchInput
-                        style={GRAPHQL_SEARCH_INPUT_STYLE}
-                        search={search}
                         showArrow
-                        mode={mode}
-                        value={selectedKeys}
+                        {...gqlSelectProps}
+                        value={selectedKeys as any}
```
Comment (dkoviazin) [inline apps/condo/domains/common/components/Table/Filters.tsx:198]: why any?
image

File: apps/condo/domains/common/components/Table/Filters.tsx [hunk 4]
```diff
@@ -1,40 +1,63 @@
...
 export const getFilterIcon: FilterIconType = (filtered) => <FilterFilled style={{ color: filtered ? colors.sberPrimary[5] : undefined }} />
 export const getFilterValue: FilterValueType = (path, filters) => get(filters, path, null)
 
-export const getTextFilterDropdown = (placeholder: string, containerStyles?: CSSProperties) => {
+export const getTextFilterDropdown: GetTextFilterDropdownType = ({ containerStyles, inputProps } = {}) => {
     return ({ setSelectedKeys, selectedKeys, confirm, clearFilters }) => {
         const handleClear = useCallback(() => {
             isFunction(clearFilters) && clearFilters()
-            setSelectedKeys('')
+            setSelectedKeys('' as any)
```
Comment (akarjakin) [inline apps/condo/domains/common/components/Table/Filters.tsx:55]: please don't use `any`, this method requires argument of type `React.Key[]` which is `type Key = string | number`

File: apps/condo/domains/common/components/Table/Filters.tsx [hunk 5]
```diff
@@ -1,40 +1,63 @@
...
             confirm({ closeDropdown: true })
         }, [clearFilters, confirm, setSelectedKeys])
 
-        const handleChangeInput = useCallback((event) => {
-            setSelectedKeys(event.target.value)
+        const handleChangeInput: CustomInputProps['onChange'] = useCallback((event) => {
+            setSelectedKeys(event.target.value as any)
```
Comment (akarjakin) [inline apps/condo/domains/common/components/Table/Filters.tsx:60]: same here

File: apps/condo/domains/common/components/Table/Filters.tsx (hunk 2 above)
Comment (akarjakin) [inline apps/condo/domains/common/components/Table/Filters.tsx:72]: `any` is bad practice, please detect exact what type should be propagated here.

File: apps/condo/domains/common/components/Table/Filters.tsx [hunk 6]
```diff
@@ -84,29 +137,33 @@ export const getOptionFilterDropdown = (options: Array<OptionType>, loading: boo
...
     return ({ setSelectedKeys, selectedKeys, confirm, clearFilters }) => {
         const handleClear = useCallback(() => {
             isFunction(clearFilters) && clearFilters()
             setSelectedKeys([])
             confirm({ closeDropdown: true })
-        }, [clearFilters])
+        }, [clearFilters, confirm, setSelectedKeys])
+
+        const handleChange: CustomSelectProps<ValueType>['onChange'] = useCallback((value) => {
+            setSelectedKeys(value as any)
```
Comment (akarjakin) [inline apps/condo/domains/common/components/Table/Filters.tsx:149]: same here

File: apps/condo/domains/common/components/Table/Filters.tsx [hunk 7]
```diff
@@ -84,29 +137,33 @@ export const getOptionFilterDropdown = (options: Array<OptionType>, loading: boo
...
-                    style={DROPDOWN_SELECT_STYLE}
-                    value={selectedKeys}
                     optionFilterProp='label'
-                    onChange={(e) => {
-                        setSelectedKeys(e)
-                        confirm({ closeDropdown: false })
-                    }}
                     {...selectProps}
+                    style={DROPDOWN_SELECT_STYLE}
+                    value={selectedKeys as any}
```
Comment (akarjakin) [inline apps/condo/domains/common/components/Table/Filters.tsx:165]: same here

File: apps/condo/domains/common/components/Table/Filters.tsx [hunk 8]
```diff
@@ -115,21 +172,16 @@ export const getSelectFilterDropdown = (selectProps: ComponentProps<typeof Selec
...
-    search: (client: ApolloClient<Record<string, unknown>>, queryArguments: string) => Promise<Array<Record<string, unknown>>>,
-    mode?: ComponentProps<typeof GraphQlSearchInput>['mode'],
-    containerStyles?: CSSProperties
-) => {
+export const getGQLSelectFilterDropdown: GetGQLSelectFilterDropdownType = ({ gqlSelectProps, containerStyles } = {}) => {
     return ({ setSelectedKeys, selectedKeys, confirm, clearFilters, visible }) => {
         const handleClear = useCallback(() => {
             isFunction(clearFilters) && clearFilters()
-            setSelectedKeys('')
+            setSelectedKeys('' as any)
```
Comment (akarjakin) [inline apps/condo/domains/common/components/Table/Filters.tsx:179]: same here

File: apps/condo/domains/common/components/Table/Filters.tsx [hunk 9]
```diff
@@ -115,21 +172,16 @@ export const getSelectFilterDropdown = (selectProps: ComponentProps<typeof Selec
...
             confirm({ closeDropdown: true })
         }, [clearFilters, confirm, setSelectedKeys])
 
-        const handleChange = useCallback((e) => {
-            setSelectedKeys(e)
+        const handleChange: ISearchInputProps['onChange'] = useCallback((value) => {
+            setSelectedKeys(value as any)
```
Comment (akarjakin) [inline apps/condo/domains/common/components/Table/Filters.tsx:184]: same here

File: apps/condo/domains/common/components/Table/Filters.tsx (hunk 3 above)
Comment (akarjakin) [inline apps/condo/domains/common/components/Table/Filters.tsx:198]: same here

File: apps/condo/domains/common/components/Table/Filters.tsx [hunk 10]
```diff
@@ -141,75 +193,71 @@ export const getGQLSelectFilterDropdown = (
...
-export const getDateFilterDropdown = (containerStyles?: CSSProperties) => {
+export const getDateFilterDropdown: GetDateFilterDropdownType = ({ datePickerProps: outerPickerProps, containerStyles } = {}) => {
     return ({ setSelectedKeys, selectedKeys, confirm, clearFilters }) => {
-        const pickerProps = {
+        const innerPickerProps: PickerProps<Dayjs> = useMemo(() => ({
             value: undefined,
-            onChange: e => {
-                setSelectedKeys(e.toISOString())
+            onChange: value => {
+                setSelectedKeys(value.toISOString() as any)
```
Comment (akarjakin) [inline apps/condo/domains/common/components/Table/Filters.tsx:213]: same here

File: apps/condo/domains/common/components/Table/Filters.tsx [hunk 11]
```diff
@@ -141,75 +193,71 @@ export const getGQLSelectFilterDropdown = (
...
                 confirm({ closeDropdown: true })
             },
             allowClear: false,
-        }
+        }), [confirm, setSelectedKeys])
 
         if (selectedKeys && selectedKeys.length > 0) {
-            pickerProps.value = dayjs(selectedKeys)
+            innerPickerProps.value = dayjs(selectedKeys as any)
```
Comment (akarjakin) [inline apps/condo/domains/common/components/Table/Filters.tsx:220]: same here

**Review by dkoviazin — APPROVED** [review]

File: apps/condo/domains/common/components/Table/Filters.tsx (hunk 3 above)
Comment (Alllex202) [reply apps/condo/domains/common/components/Table/Filters.tsx:198]: type Key = string | number

selectedKeys have type Key[]

type Key[] cannot cast to type string[]

Comment (Alllex202) [reply apps/condo/domains/common/components/Table/Filters.tsx]: changed to @ts-ignore and added note about type casting problem

File: apps/condo/domains/common/components/Table/Filters.tsx (hunk 4 above)
Comment (Alllex202) [reply apps/condo/domains/common/components/Table/Filters.tsx]: changed to @ts-ignore and added note about type casting problem

**Comment by Alllex202** [comment]
The problem of type casting can be solved by adding a type for filtering in a column and getting values based on this type.

Probably should come back to this issue later, as it's way out of the scope of the current task and doesn't have any negative impact

**Pull request merged**